bin/src/main.rs

```diff
@@ -178,6 +180,82 @@ async fn main() {
         }
     };
 
+    let k8s_event_stream_client = k8s_event_stream.as_ref().unwrap().client.clone();
+    let mut k8s_claimed_lease: Option<String> = None;
+    match &k8s_event_stream {
+        Some(stream) => {
+            if &config.startup.option == "on" {
+                // Attempt to claim lease N times, then move on.
+                info!("Getting agent-startup-lease (making limited attempts)");
+                let k8s_lease_api = k8s::lease::get_k8s_lease_api(
+                    &std::env::var("NAMESPACE").unwrap(),
+                    stream.client.clone(),
+                )
+                .await;
+                for i in 0..K8S_STARTUP_LEASE_RETRY_ATTEMPTS {
+                    info!("Attempting connection: {}", i);
+                    match get_available_lease(K8S_STARTUP_LEASE_LABEL, &k8s_lease_api).await {
+                        Some(available_lease) => {
+                            info!("Lease available: {:?}", available_lease);
+                            k8s::lease::claim_lease(
+                                available_lease,
+                                std::env::var("POD_NAME").unwrap(),
+                                &k8s_lease_api,
+                                &mut k8s_claimed_lease,
+                            )
+                            .await;
+                            break;
+                        }
+                        None => {
+                            info!("No lease availabe at this time. Waiting 1 second...");
+                            thread::sleep(Duration::from_millis(1000));
```

Comment:
We definitely don't want to use thread::sleep here, tokio has it's own async sleep mechanism which we should use

Reply:
Updated to use tokio.